hilti/toolchain/src/compiler/cxx/linker.cc

```diff
@@ -49,19 +49,25 @@ void cxx::Linker::finalize() {
     // Create the HLTO version information.
     auto version = rt::library::Version{.magic = "v1",
                                         .hilti_version = configuration().version_number,
-                                        .created = rt::time::current_time().seconds(),
                                         .debug = _codegen->context()->options().debug,
                                         .optimize = _codegen->context()->options().optimize};
 
-    unit.add(cxx::declaration::IncludeFile{"hilti/rt/libhilti.h"});
     for ( const auto& p : plugin::registry().plugins() )
         for ( const auto& i : p.cxx_includes )
             unit.add(cxx::declaration::IncludeFile{i});
 
     unit.add(fmt("const char HILTI_EXPORT HILTI_WEAK * __hlto_library_version = R\"(%s)\";", version.toJSON()));
 
     // Create a scope string that's likely to be unique to this linker module.
-    auto scope = hilti::rt::fmt("%" PRIx64, hilti::rt::time::current_time().nanoseconds() % 0xffffff);
+    std::size_t hash = 0;
+    for ( const auto& [id, path] : _modules ) {
+        std::ifstream ifs(path);
+        std::string content((std::istreambuf_iterator<char>(ifs)), (std::istreambuf_iterator<char>()));
+        auto h = std::hash<std::string>()(content);
+        hash = hash ^ (h << 1);
+    }
```

Comment:
This doesn't seem to get the job done for Linux and might need another approach. The motivation here is to remove the time overhead from having to compile the `__linker__*.cc` files, here by making it cachable via `ccache`. Compiling these files is expensive since it requires parsing all of `libhilti.h`, even if unused for JIT without joins.